src/search/algorithms.clj

```diff
@@ -63,6 +63,35 @@
             new-visited (clojure.set/union children visited)]
         (recur new-frontier new-came-from new-visited)))))
 
+(defn a-star-search [children-fn cost-fn heuristic-fn start-state goal-state & {:keys [max-states] :or {max-states 1000000}}]
+  (loop [max-states max-states
+        frontier (pm/priority-map start-state 0)
+         came-from {}
+         cost-so-far {start-state 0}]
+    (if (or (neg? max-states)
+            (empty? frontier)
+            (>= (count came-from) max-states)
```

Comment:
I'm not sure why this is here. You're reducing `max-states` every time through the loop, and you check to see if it's negative two lines earlier, so do we need this check?